src/plugins/discover/public/application/angular/discover.js

```diff
@@ -896,6 +929,14 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
       });
     });
 
+    if ($scope.useNewFieldsApi) {
+      // get some columns to display
+      if ($scope.state.columns.length === 0) {
+        const columns = ['fields'];
+        setAppState({ columns });
```

Comment:
Don't think `fields` should be added to state, think it's better that way: if columns of state are empty -> display fields (or _source for the legacy way)

Reply:
Fair point, but can we display them without adding them to state?

Reply:
e.g. just by using a simple `getDisplayColumns` helper function that returns the default columns, if state is empty

Reply:
@majagrubic there's a problem with the current handling of `fields`, it should be considered as a fallback, just something that is displayed, but e.g. when I add an index pattern that actually has a field named `fields`

<img width="1661" alt="Discover_-_Elastic" src="https://user-images.githubusercontent.com/463851/101011770-fb0e0800-3562-11eb-9a21-35fd08b7d060.png">

Reply:
@majagrubic now I can add and remove `fields`, but when I add it, it is expected to display : "magnetic", but the application built _source replacement is displayed:

<img width="1113" alt="Bildschirmfoto 2020-12-07 um 10 01 35" src="https://user-images.githubusercontent.com/463851/101331220-f6539780-3873-11eb-9bc2-f70af2fe2261.png">

Here's what I've been using:

```
PUT fields/_doc/1 
{
  "fields": "magnetic"
}

```
created an index patteren, added `fields` to selected fields

Reply:
yeah I'd second that we should treat `fields` specially and not as a column. I even think in contrast to how we handle `_source` at the moment, we even don't need to show any "selected fields" in the sidebar in this case. So we could really treat it as "no columns" -> just show all fields in the table, but the sidebar has nothing under selected fields in this case, also meaning, you can not add "fields" as a dedicated column when there are other columns, which I think we wouldn't want.

Reply:
was resolved in the meantime